packages/plugin-betswirl/src/actions/coinToss.ts

```diff
@@ -0,0 +1,354 @@
+import { z } from "zod";
+import { type Hex } from "viem";
+import {
+    type Action,
+    type IAgentRuntime,
+    type Memory,
+    type HandlerCallback,
+    type State,
+    elizaLogger,
+    type ActionExample,
+    composeContext,
+    generateObject,
+    ModelClass,
+} from "@moxie-protocol/core";
+import { MoxieWalletClient } from "@moxie-protocol/moxie-agent-lib/src/wallet";
+import {
+    CASINO_GAME_TYPE,
+    CoinToss,
+    COINTOSS_FACE,
+    slugById,
+    formatTxnUrl,
+} from "@betswirl/sdk-core";
+import { casinoBetParams, getMaxBetCountParam } from "../types";
+import {
+    getChainIdFromWallet,
+    getBetToken,
+    placeBet,
+    getBet,
+    getBetAmountInWei,
+} from "../utils/betswirl";
+import { formatTokenForMoxieTerminal } from "../utils/moxie";
+
+export const CoinTossBetParameters = z.object({
+    face: z
+        .nativeEnum(COINTOSS_FACE)
+        .nullable()
+        .describe("The face of the coin"),
+    ...casinoBetParams,
+    ...getMaxBetCountParam(CASINO_GAME_TYPE.COINTOSS),
```

Comment:
type validation for `isConfirmed` to indicate confirmation from users:
```suggestion
    ...getMaxBetCountParam(CASINO_GAME_TYPE.COINTOSS),
    isConfirmed: z
        .boolean()
        .optional()
        .nullable()
        .describe(
            "Whether the user confirmed the bet based on historical conversation."
        ),
```